src/test/java/htsjdk/samtools/filter/IntervalKeepPairFilterTest.java

```diff
@@ -0,0 +1,114 @@
+package htsjdk.samtools.filter;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.util.CollectionUtil;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import htsjdk.samtools.util.Interval;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.StreamSupport;
+
+public class IntervalKeepPairFilterTest {
+    private static final int READ_LENGTH = 151;
+    private final SAMRecordSetBuilder builder = new SAMRecordSetBuilder();
+
+    @BeforeTest
+    public void setUp() {
+        builder.setReadLength(READ_LENGTH);
+        //should be kept in first test, filtered out in third
+        builder.addPair("mapped_pair_chr1", 0, 1, 151);
+        //should be filtered out for first test, and kept in third
+        builder.addPair("mapped_pair_chr2", 1, 1, 151);
+        //first read should pass, second should not, but both will be kept in first test
+        builder.addPair("one_of_pair", 0, 1, 1000);
+        //second read unmapped but both should be kept in test 4 because first read overlaps interval
```

Comment:
instead of `test 4` describe the test: `...should be kept when the first read overlaps the interval (is on chr4)` or something like it.  You may add more test cases and then this doc gets out of date.